packages/langium/src/grammar/lsp/langium-grammar-reference-finder.ts

```diff
@@ -0,0 +1,196 @@
+
+    getInterfacesFromAssignment(ref: AbstractType): Set<Interface> {
+        let interfaces: Set<Interface> = new Set<Interface>();
+        if (isInterface(ref)) {
+            interfaces = this.collectSuperTypes(ref);
+            interfaces.add(ref);
+        } else if (isType(ref)) {
+            interfaces = this.collectSuperTypes(ref);
+        }
+        return interfaces;
+    }
+
+    getReferencesFromTypeAttribute(typeAttributeNode: TypeAttribute, selectedNode: LeafCstNode): Array<{ docUri: URI; range: Range; }> {
+        const refs: Array<{ docUri: URI; range: Range; }> = [];
+        const interfaceNode = getContainerOfType(typeAttributeNode, isInterface);
+        if (interfaceNode) {
+            const collectedTypes = this.collectChildrenTypes(interfaceNode);
+            collectedTypes.add(interfaceNode);
+
+            let referencesToTypes: Set<ReferenceDescription> = new Set<ReferenceDescription>();
+            collectedTypes.forEach(collectedType => {
+                const refs = this.references.findReferences(collectedType).toSet();
+                referencesToTypes = new Set<ReferenceDescription>([...referencesToTypes, ...refs]);
+            });
+
+            const parserRules: Set<ParserRule | Action> = this.findParserRulesFromReferences(referencesToTypes);
+
+            parserRules.forEach(rule => {
+                const assignment = isParserRule(rule) ? extractAssignments(rule.alternatives).find(assignment => assignment.feature === selectedNode.text) : extractAssignments(getContainerOfType(rule, isGroup)!).find(assignment => assignment.feature === selectedNode.text);
+
+                if (assignment) {
+                    const cstNode = assignment.$cstNode;
+                    if (cstNode) {
+                        const leaf = findLeafNodeAtOffset(cstNode, cstNode.offset);
+                        if (leaf) {
+                            const rootNode = extractRootNode(leaf.element);
+                            if (rootNode) {
+                                refs.push({docUri: rootNode.$document!.uri, range: leaf.range});
+                            }
+                        }
+                    }
+                }
+            });
+        }
+        return refs;
+    }
+
+    findParserRulesFromReferences(referencesToTypes: Set<ReferenceDescription>): Set<ParserRule | Action> {
+        const parserRules: Set<ParserRule | Action> = new Set<ParserRule | Action>();
+        referencesToTypes.forEach(ref => {
+            const doc = this.langiumDocuments.getOrCreateDocument(ref.sourceUri);
+            const astNode = this.astNodeLocator.getAstNode(doc, ref.sourcePath);
+            if (isParserRule(astNode) || isAction(astNode)) {
+                parserRules.add(astNode);
+            }
+        });
+        return parserRules;
+    }
+
+    collectChildrenTypes(interfaceRule: Interface): Set<Interface | Type> {
+        let childrenTypes: Set<Interface | Type> = new Set<Interface | Type>();
```

Comment:
Minor: Type can be inferred
```suggestion
        let childrenTypes = new Set<Interface | Type>();
```